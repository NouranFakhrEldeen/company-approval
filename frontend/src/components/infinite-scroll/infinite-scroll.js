import React from 'react';
import InfiniteScroll from 'react-infinite-scroll-component';
import { withTranslation } from 'react-i18next';
import NoResults from '../../components/noResults/noResults';
// import Moment from 'react-moment';

const infiniteScrollComponent = ({
  next,
  dataLength,
  hasMore,
  loader,
  onClick,
  previewed,
  no_results,

}) => {

  return (
    <div className={
      !previewed?.length ? 'no-results-div' : ''}>
      <InfiniteScroll
        next={next}
        dataLength={dataLength}
        hasMore={hasMore}
        loader={loader}>

        <div className='list-group striped' id='results-container'>
          {
            !previewed?.length
              ?
              <NoResults value={no_results} />
              :
              previewed?.map((item, index) => (
                <div
                  onClick={() => onClick(item.realItem)}
                  key={index}
                  className='list-group-item list-group-item-with-actions cursorPointer'>
                  <p>{item?.header}</p>
                  <span>
                    {item?.body}
                  </span>
                </div>
              ))
          }
        </div>
      </InfiniteScroll>
    </div>
  );
};

export const InfiniteScrollComponent = (withTranslation()(infiniteScrollComponent));
// export const InfiniteScrollComponent = infiniteScrollComponent;

