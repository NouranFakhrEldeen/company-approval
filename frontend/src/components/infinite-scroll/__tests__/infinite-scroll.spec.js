import React from 'react';

// import renderer from 'react-test-renderer';
import { InfiniteScrollComponent } from '../infinite-scroll';
import { render } from '@testing-library/react';
import { HashRouter as Router } from 'react-router-dom';

describe('should test infinite scroll', () => {

  let cars = [
    {
      'color': 'purple',
      'name': 'minivan',
      'businessId': 12345,
      'item': 7,
    },
    {
      'color': 'red',
      'name': 'station wagon',
      'businessId': 345456,
      'item': 5,
    },
  ];

  // snapshot

  it('should do snapshot', () => {
    /*const ButtonComp = renderer.create(
      <InfiniteScrollComponent
        dataLength={cars?.length}
        previewed={[...cars.map(item => ({
          createdAt: '2020-01-01',
          header: item?.name,
          body: item.businessId,
          realItem: item,
        }))]}

      />
    ).toJSON();
    expect(ButtonComp).toMatchSnapshot();*/
    const { asFragment } = render(
      <Router>
        <InfiniteScrollComponent
          dataLength={cars?.length}
          previewed={[...cars.map(item => ({
            createdAt: '2021-03-28T08:16:54.277Z',
            header: item?.name,
            body: item.businessId,
            realItem: item,
          }))]}

        />
      </Router>,
    );
    const firstRender = asFragment();
    expect(firstRender).toMatchSnapshot();
  });
});
