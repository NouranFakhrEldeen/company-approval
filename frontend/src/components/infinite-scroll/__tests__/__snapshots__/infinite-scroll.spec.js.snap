// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test infinite scroll should do snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <div
      class="infinite-scroll-component__outerdiv"
    >
      <div
        class="infinite-scroll-component "
        style="height: auto; overflow: auto;"
      >
        <div
          class="list-group striped"
          id="results-container"
        >
          <div
            class="list-group-item list-group-item-with-actions cursorPointer"
          >
            <p>
              minivan
            </p>
            <span>
              12345
            </span>
          </div>
          <div
            class="list-group-item list-group-item-with-actions cursorPointer"
          >
            <p>
              station wagon
            </p>
            <span>
              345456
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
